kratos/elements/distance_calculation_flux_based_element.h

```diff
@@ -0,0 +1,359 @@
+Currently only supports very viscous problems (polymers or RTM)
+It requires two steps to be performed:
+1. Solves a transient diffusion problem to get a field equivalent to a potential.
+2. from the field, compute gradients to obtain velocity and use it to compute flowlenght/filltime.
+*/
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template< unsigned int TDim,  unsigned int TNumNodes >
+class KRATOS_API(KRATOS_CORE) DistanceCalculationFluxBasedElement : public Element
+{
+public:
+
+    /// Counted pointer of
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(DistanceCalculationFluxBasedElement);
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Node<3> NodeType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef Geometry<NodeType>::PointsArrayType NodesArrayType;
+    typedef Vector VectorType;
+    typedef Matrix MatrixType;
+    typedef std::size_t IndexType;
+    typedef std::size_t SizeType;
+    typedef std::vector<std::size_t> EquationIdVectorType;
+    typedef std::vector< Dof<double>::Pointer > DofsVectorType;
+    typedef PointerVectorSet<Dof<double>, IndexedObject> DofsArrayType;
+    typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
+
+    static constexpr unsigned int Dim = TDim;
+
+    static constexpr unsigned int NumNodes = TNumNodes;
+
+    static constexpr unsigned int BlockSize = 1;
+
+    static constexpr unsigned int LocalSize = NumNodes * BlockSize;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constuctor.
+    /**
+    * @param NewId Index number of the new element (optional)
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId = 0) :
+        Element(NewId)
+    {}
+
+    /// Constructor using an array of nodes.
+    /**
+    * @param NewId Index of the new element
+    * @param ThisNodes An array containing the nodes of the new element
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, const NodesArrayType& ThisNodes):
+        Element(NewId, ThisNodes)
+    {}
+
+    /// Constructor using a geometry object.
+    /**
+    * @param NewId Index of the new element
+    * @param pGeometry Pointer to a geometry object
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, GeometryType::Pointer pGeometry):
+        Element(NewId, pGeometry)
+    {}
+
+    /// Constuctor using geometry and properties.
+    /**
+    * @param NewId Index of the new element
+    * @param pGeometry Pointer to a geometry object
+    * @param pProperties Pointer to the element's properties
+    */
+    DistanceCalculationFluxBasedElement(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties) :
+        Element(NewId, pGeometry, pProperties)
```

Comment:
```suggestion
    DistanceCalculationFluxBasedElement(
        IndexType NewId,
        GeometryType::Pointer pGeometry,
        Properties::Pointer pProperties)
        : Element(NewId, pGeometry, pProperties)
```
Style guide suggestion. Please do so in the other constructors.

Reply:
ok